detectable/src/main/java/com/synopsys/integration/detectable/detectables/ivy/parse/IvyParseExtractor.java

```diff
@@ -0,0 +1,51 @@
+package com.synopsys.integration.detectable.detectables.ivy.parse;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Optional;
+
+import javax.xml.parsers.SAXParser;
+
+import com.synopsys.integration.bdio.graph.MutableMapDependencyGraph;
+import com.synopsys.integration.bdio.model.dependency.Dependency;
+import com.synopsys.integration.bdio.model.externalid.ExternalIdFactory;
+import com.synopsys.integration.detectable.detectable.codelocation.CodeLocation;
+import com.synopsys.integration.detectable.detectables.ivy.IvyProjectNameParser;
+import com.synopsys.integration.detectable.extraction.Extraction;
+import com.synopsys.integration.util.NameVersion;
+
+public class IvyParseExtractor {
+    private final ExternalIdFactory externalIdFactory;
+    private final SAXParser saxParser;
+    private final IvyProjectNameParser projectNameParser;
+
+    public IvyParseExtractor(ExternalIdFactory externalIdFactory, SAXParser saxParser, IvyProjectNameParser projectNameParser) {
+        this.externalIdFactory = externalIdFactory;
+        this.saxParser = saxParser;
+        this.projectNameParser = projectNameParser;
+    }
+
+    public Extraction extract(File ivyXmlFile, File buildXmlFile) {
+        try (InputStream ivyXmlInputStream = new FileInputStream(ivyXmlFile)) {
+            IvyDependenciesHandler ivyDependenciesHandler = new IvyDependenciesHandler(externalIdFactory);
+            saxParser.parse(ivyXmlInputStream, ivyDependenciesHandler);
+            List<Dependency> dependencies = ivyDependenciesHandler.getDependencies();
+
+            MutableMapDependencyGraph dependencyGraph = new MutableMapDependencyGraph();
+            dependencyGraph.addChildrenToRoot(dependencies);
+
+            CodeLocation codeLocation = new CodeLocation(dependencyGraph);
+
+            Optional<NameVersion> projectName = projectNameParser.parseProjectName(buildXmlFile);
+
+            return new Extraction.Builder()
+                .success(codeLocation)
+                .nameVersionIfPresent(projectName)
+                .build();
+        } catch (Exception e) {
+            return new Extraction.Builder().exception(e).build();
+        }
```

Comment:
Jordan and I recently discussed this.
If you can provide additional context to the failure, then catch the exceptions like you've done here.
If there is no additional context to provide, then Detect can handle the boilerplate code. 
See the comment above this method. 
https://github.com/blackducksoftware/synopsys-detect/blob/5c9a12d98eaa67473f05c485a61ea0c327dfd64f/detectable/src/main/java/com/synopsys/integration/detectable/Detectable.java#L32

Reply:
This code throws an IOException and a SAXException.  Detectable handles IOException, so I can just add that to the IvyParseExtractor's method signature.  Detectable does not handle SAXException, though.  Does it make more sense to leave the catch block and change the catch type to SAXException, or to modify Detectable's method signature?